kraken-build/src/kraken/std/python/tasks/mypy_task.py

```diff
@@ -49,11 +47,11 @@ def get_execute_command_v2(self, env: MutableMapping[str, str]) -> list[str]:
         status_file = (self.project.directory / ".dmypy.json").absolute()
         if use_daemon:
             command += ["--status-file", str(status_file), "run", "--"]
-        if mypy_pex_bin:
-            # Have mypy pick up the Python executable from the virtual environment that is activated automatically
-            # during the execution of this task as this is an "EnvironmentAwareDispatchTask". If we don't supply this
-            # option, MyPy will only know the packages in its PEX.
-            command += ["--python-executable", "python"]
+        if mypy_cmd and self.settings.build_system is not None:
+            # Have mypy pick up the Python executable from the managed virtual environment for this project.
+            # If we don't supply this, MyPy will only know the packages in its own virtual environment.
+            managed_env_path = self.settings.build_system.get_managed_environment().get_path()
+            command += ["--python-executable", os.fspath(managed_env_path / "bin" / "python")]
```

Comment:
Another solution might be to specify the python interpreter within the venv dedicated to the tool. Something along this line `uv tool run --python /path/to/venv/bin/python mypy`.

Reply:
I don't expect that'll work, as the `uv tool run` entrypoint still creates a new virtual environment using that interpreter, which won't have the project dependencies installed in it: https://github.com/astral-sh/uv/blob/ec480bd3ee43da2f5f15e7c6383c334e78894c31/crates/uv/src/commands/tool/run.rs#L118